packages/cpt-ui/components/EpsRoleSelectionPage.tsx

```diff
@@ -81,90 +81,96 @@ export default function RoleSelectionPage({ contentText }: RoleSelectionPageProp
-                    rolesWithAccess.map((role: RoleDetails, index: number) => ({
-                        role,
-                        link: "/yourselectedrole",
-                        uuid: `{role_with_access_${index}}`,
-                    }))
+            if (response.status !== 200) {
+                throw new Error(
+                    `Server did not return CPT user info, response ${response.status}`
                 )
+            }
 
-                setRolesWithoutAccess(
-                    rolesWithoutAccess.map((role: RoleDetails, index: number) => ({
-                        uuid: `{role_without_access_${index}}`,
-                        roleName: role.role_name || noRoleName,
-                        orgName: role.org_name || noOrgName,
-                        odsCode: role.org_code || noODSCode,
-                    }))
-                )
+            const data = await response.json()
 
-                // If the user has exactly one accessible role and zero roles without access,
-                // redirect them immediately
-                if (rolesWithAccess.length === 1 && rolesWithoutAccess.length === 0) {
-                    setRedirecting(true);
-                    setSelectedRole(rolesWithAccess[0]);
-                    router.push("/searchforaprescription");
-                    return;
-                }
-            })
-            .catch((err) => {
-                setError("Failed to fetch CPT user info");
-                console.error("Error fetching tracker user info:", err);
-            })
-            .finally(() => {
-                setLoading(false);
-            });
+            if (!data.userInfo) {
+                throw new Error("Server response did not contain data")
+            }
+
+            const userInfo: TrackerUserInfo = data.userInfo
+
+            const rolesWithAccess = userInfo.roles_with_access
+            const rolesWithoutAccess = userInfo.roles_without_access
+            const currentlySelectedRole = userInfo.currently_selected_role ? {
+                ...userInfo.currently_selected_role,
+                uuid: `selected_role_0`
+            } : undefined
+
+            // Populate the EPS card props
+            setRolesWithAccess(
+                rolesWithAccess.map((role: RoleDetails, index: number) => ({
+                    uuid: `{role_with_access_${index}}`,
+                    role,
+                    link: "/yourselectedrole"
+                }))
+            )
+
+            setRolesWithoutAccess(
+                rolesWithoutAccess.map((role: RoleDetails, index: number) => ({
+                    uuid: `{role_without_access_${index}}`,
+                    roleName: role.role_name ? role.role_name : noRoleName,
+                    orgName: role.org_name ? role.org_name : noOrgName,
+                    odsCode: role.org_code ? role.org_code : noODSCode
+                }))
+            )
+
+            setCurrentlySelectedRole(currentlySelectedRole)
```

Comment:
You'll want to update to the `setSelectedRole` state in the `AccessProvider`, since that's the context that carries the selected role around with it on other pages! It's already imported by the look of things, so you only need to replace the state local to this component with that one